src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -77,14 +79,48 @@ private async Task<Document> UseCollectionDirectly(Document document, TextSpan s
             }
 
             var generator = SyntaxGenerator.GetGenerator(document);
-            var indexNode = generator.LiteralExpression(0);
-            var elementAccessNode = generator.ElementAccessExpression(collectionSyntax.WithoutTrailingTrivia(), indexNode)
-                .WithTrailingTrivia(invocationNode.GetTrailingTrivia());
 
-            var newRoot = root.ReplaceNode(invocationNode, elementAccessNode);
+            var elementAccessNode = GetReplacementNode(generator, collectionSyntax, methodName);
+            if (elementAccessNode == null)
+            {
+                return document;
+            }
+
+            var newRoot = root.ReplaceNode(invocationNode, elementAccessNode.WithTrailingTrivia(invocationNode.GetTrailingTrivia()));
             return document.WithSyntaxRoot(newRoot);
         }
 
+        private static SyntaxNode GetReplacementNode(SyntaxGenerator generator, SyntaxNode collectionSyntax, string methodName)
+        {
+            var collectionSyntaxNoTrailingTrivia = collectionSyntax.WithoutTrailingTrivia();
+
+            if (methodName == "First")
+            {
+                var zeroLiteral = generator.LiteralExpression(0);
+                return generator.ElementAccessExpression(collectionSyntaxNoTrailingTrivia, zeroLiteral);
+            }
+
+            if (methodName == "Last")
+            {
+                // TODO: Handle C# 8 index expression (and vb.net equivalent if any)
```

Comment:
I don't know how to detect that the target code supports the index expression syntax. I have seen that I could access to `semanticModel.Compilation.LanguageVersion` if I cast the compilation to the C# or VB one but that seems wrong.

Reply:
I think it is fine to start with your current implementation. VS2019 should offer a subsequent refactoring to convert it to an index expression.

Reply:
Shall I remove the todo comment? Shall I create some separate ticket?